### PR TITLE
LibJS: Misc String prototype fixes

### PR DESCRIPTION
**Pull request by IdanHo**
This fixes ~35 test262 test cases.

### CODE REVIEW

File: Userland/Libraries/LibJS/Runtime/StringPrototype.cpp [hunk 1]
```diff
@@ -711,6 +712,44 @@ JS_DEFINE_NATIVE_FUNCTION(StringPrototype::match)
...
+            auto flags_string = flags_object.to_string(global_object);
+            if (vm.exception())
+                return {};
+            if (!flags_string.contains("g")) {
+                vm.throw_exception<TypeError>(global_object, ErrorType::StringMatchAllNonGlobalRegExp);
+                return {};
+            }
+        }
+        if (auto* matcher = get_method(global_object, regexp, vm.well_known_symbol_match_all()))
+            return vm.call(*matcher, regexp, this_object);
```
Comment (linusg) [inline Userland/Libraries/LibJS/Runtime/StringPrototype.cpp:743]: Missing exception check after this

File: Userland/Libraries/LibJS/Runtime/StringPrototype.cpp [hunk 2]
```diff
@@ -800,6 +801,26 @@ JS_DEFINE_NATIVE_FUNCTION(StringPrototype::replace)
...
+// 22.1.3.19 String.prototype.search ( regexp ), https://tc39.es/ecma262/#sec-string.prototype.search
+JS_DEFINE_NATIVE_FUNCTION(StringPrototype::search)
+{
+    auto this_object = require_object_coercible(global_object, vm.this_value(global_object));
+    if (vm.exception())
+        return {};
+    auto regexp = vm.argument(0);
+    if (!regexp.is_nullish()) {
+        if (auto* searcher = get_method(global_object, regexp, vm.well_known_symbol_search()))
+            return vm.call(*searcher, regexp, this_object);
```
Comment (linusg) [inline Userland/Libraries/LibJS/Runtime/StringPrototype.cpp:813]: Missing exception check after this

File: Userland/Libraries/LibJS/Runtime/StringPrototype.cpp (hunk 1 above)
Comment (IdanHo) [reply Userland/Libraries/LibJS/Runtime/StringPrototype.cpp:743]: added

File: Userland/Libraries/LibJS/Runtime/StringPrototype.cpp (hunk 2 above)
Comment (IdanHo) [reply Userland/Libraries/LibJS/Runtime/StringPrototype.cpp:815]: added

**Pull request merged**